docs/api.md

```diff
@@ -118,8 +120,93 @@ type AddonsSpec struct {
 }
 
 // TODO maybe we have profiles for scaling behaviors
-type AutoScalerAddonSpec struct {
-    ResourceLimits LimitRange
+type ClusterAutoscaling struct {
```

Comment:
General note: bool fields in API are generally suspicious, and I also wonder how many of these are often set by users and which might be considered esoteric and could be omitted from the minimum viable first draft of the API.

Reply:
i've left it in for now, but i think in the specific case of the `ScaleDownEnabled` (new name added in recent push), we could probably evaluate just dropping it completely.

i am curious to hear more about your thoughts behind 
> bool fields in API are generally suspicious
we use this bool in the CAO, and if it's bad form i'd like to evaluate if we can make this easier there as well.